game/core/src/main/java/com/gg/game/Model/EnemyActionHandler.java

```diff
@@ -0,0 +1,70 @@
+package com.gg.game.Model;
+
+import java.util.List;
+
+/**
+ * Class for managing behavior of enemies in a room.
+ * Singleton.
+ */
+class EnemyActionHandler {
+
+    private static volatile EnemyActionHandler instance;
+    private Room currentRoom;
+    //private Vector targetPos;
+
+    private EnemyActionHandler(){
+        if(instance != null){
+            throw new RuntimeException("Use getInstance() method to get the single instance of this class");
+        }
+    }
+
+    public static EnemyActionHandler getInstance(){
+        if (instance == null){ //If there is no instance available, create a new one.
+            instance = new EnemyActionHandler();
+        }
+        return instance;
+    }
+
+    /**
+     * Performs all actions of enemies.
+     * @param targetPos which target to track.
+     */
+    public void updateEnemies(Vector targetPos){
+        moveEnemies(targetPos);
+        //enemies fire //TODO: add fire when projectiles and weapons exist.
+    }
+
+    /**
+     * Changes currently handled room.
+     * @param room
+     */
+    public void setCurrentRoom(Room room){
+        currentRoom=room;
```

Comment:
```suggestion
        this.currentRoom=room;
```
Increases clarity of which scope we are assigning the value to.